x-pack/plugins/security_solution/public/app/home/index.tsx

```diff
@@ -24,23 +25,31 @@ import { GlobalHeader } from './global_header';
 import { SecuritySolutionTemplateWrapper } from './template_wrapper';
 import { ConsoleManager } from '../../management/components/console/components/console_manager';
 import { useSyncGlobalQueryString } from '../../common/utils/global_query_string';
-import { useInitSearchBarUrlParams } from '../../common/hooks/search_bar/use_init_search_bar_url_params';
+import { useInitSearchBarFromUrlParams } from '../../common/hooks/search_bar/use_init_search_bar_url_params';
+import { useInitTimerangeFromUrlParam } from '../../common/hooks/search_bar/use_init_timerange_url_params';
+import { useUpdateTimerangeOnPageChange } from '../../common/hooks/search_bar/use_update_timerange_on_page_change';
 
 interface HomePageProps {
   children: React.ReactNode;
   onAppLeave: (handler: AppLeaveHandler) => void;
   setHeaderActionMenu: AppMountParameters['setHeaderActionMenu'];
 }
 
+const useUrlState = () => {
```

Comment:
Wouldn't it be better to move those hooks away from the home page, and just load a single `useUrlState` here? This way the main home component is decoupled from any urlState logic. Maybe a to a new `x-pack/plugins/security_solution/public/common/hooks/use_url_state` file?